### ISSUE TITLE
Add assembleStorage and disassembleStorage tests for DEN

### ISSUE THREAD

**Issue by KyleVaughn** [bkochuna/ners570f23-SpMV#35]
## Description:
Tests must be created to ensure the accurate conversion of sparse matrix data to a dense format, as well as to ensure correct memory handling for this task. We specifically need tests for the functions "assembleStorage(...)" and "disassembleStorage(...)".

## Tasks:
**For 'assembleStorage':**
- Create a sparse matrix with known values.
- Use the assembleStorage(...) function to convert the sparse matrix to a dense matrix.
- Validate that the resulting dense matrix matches expected values.
- Check for any memory leaks or unnecessary memory allocations during the conversion process.

**For 'disassembleStorage':**
- Start with a dense matrix with known values.
- Use the disassembleStorage(...) function to convert the dense matrix back to sparse format.
- Validate that the resulting sparse matrix matches expected sparse representation (like specific row, column indices, and values).
- Ensure no memory leaks or unnecessary memory allocations during the conversion back to sparse

## Definition of done:
This issue will be resolved when both assembleStorage and disassembleStorage tests are implemented and...
- Tests cover multiple edge cases
- Test results are documented and discrepancies are reported
- Tests are passed without memory issues

**Comment by KyleVaughn** [comment]
Review by @arsridha

**Comment by carltonjames** [comment]
@arsridha I've finished writing this issue now. Sorry about that.

**Comment by arsridha** [comment]
I don't think it is required to test if there is an accurate conversion between the dense and sparse matrix formats. I believe these tests are supposed to check whether the assemble/disassemble storage methods are defined accurately. You would also need to test if these methods fill in the dense matrix with the correct coefficients and have a brief mention of disassembling. You would also need to modify your tasks accordingly.  

As far as the description goes, maybe a couple of sentences about what these tests do and why they are required might help.

**Comment by carltonjames** [comment]
@arsridha Thank you for your feedback. I've updated the issue description accordingly. I've also implemented the tests and committed them to our pull request. Also, I removed the tasks related to checking for memory leaks since it was said in lecture that they were not required.
